### PR TITLE
feat(analytics): add more events

### LINKED ISSUE #2951: Fix unhandled rejections in unit tests
image

### PR DESCRIPTION
**Pull request by mroz22**
contribution to #2882

fix #2951

### CODE REVIEW

File: packages/suite/src/actions/suite/analyticsActions.ts [hunk 1]
```diff
@@ -311,15 +334,27 @@ export const init = (loadedState: State, optout: boolean) => (
...
     });
     // 5. if analytics was initiated as enabled, continue with setting up side effects
     if (!getState().analytics.enabled) return;
     // 6. error logging to sentry
     setSentryUser(instanceId);
-    // 7. register event listeners to report events from electron
-    // todo:
+    // 7. register event listeners
+    // @ts-ignore. how do we handle native with respect to window specific event listeners?
+    window.addEventListener('beforeunload', () => {
```
Comment (szymonlesisz) [inline packages/suite/src/actions/suite/analyticsActions.ts]: move it to utils

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

File: packages/suite/src/actions/suite/analyticsActions.ts (hunk 1 above)
Comment (mroz22) [reply packages/suite/src/actions/suite/analyticsActions.ts]: Moved it into utils, but still I don't feel 100% happy. Maybe there could be a better, more generic solution. But I would need to dive into react-native a bit first to find out how these listners are used there. If there is one global object, a counter part to window, this function could be like `setGlobalListener(event, callback)`

**Comment by mroz22** [comment]
it seems there is a race condition in e2e tests. will fix.

File: packages/suite/src/middlewares/suite/analyticsMiddleware.ts [hunk 2]
```diff
@@ -68,7 +69,8 @@ const analytics = (api: MiddlewareAPI<Dispatch, AppState>) => (next: Dispatch) =
                         payload: {
                             mode,
                             firmware: `${features.major_version}.${features.minor_version}.${features.patch_version}`,
-                            // backup_type: features.backup_type || 'Bip39', // @ts-ignore todo add to features types, missing
+                            // @ts-ignore todo add to features types in connect https://github.com/trezor/connect/pull/702
```
Comment (szymonlesisz) [inline packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: it should be fixed in new connect

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

File: packages/suite/src/middlewares/suite/analyticsMiddleware.ts (hunk 2 above)
Comment (mroz22) [reply packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: Removed :rocket:

**Review by szymonlesisz — APPROVED** [review]

**Pull request merged**